benchmarks/bench_ml.py

```diff
@@ -0,0 +1,445 @@
+
+        # apply transformation to model parameters, for example exp transformation
+        point_mapped = {}
+
+        for param, v in point.iteritems():
+            v = self.model_class[MODEL_PARAMETERS][param][PARAM_MAP](v)
+            point_mapped[param] = v
+
+        model = cls(**point_mapped)
+
+        try:
+            model.fit(X, Y)
+            r = model.score(Xv, Yv)
+        except BaseException as ex:
+            r = 0.0 # on error: return assumed smallest value of objective function
+
+        # while negative values could be informative, they could be very large also,
+        # which could mess up the optimization procedure. Suggestions are welcome.
+        r = max(r, 0.0)
+
+        return r
+
+# this is necessary to generate table for README in the end
+table_template = """|Blackbox Function| Minimum | Best minimum | Mean f_calls to min | Std f_calls to min | Fastest f_calls to min
+------------------|------------|-----------|---------------------|--------------------|-----------------------
+|ML_hypp_tune|"""
+
+def calculate_performance(all_data):
+    """
+    Calculates the performance metrics as found in "benchmarks" folder of scikit-optimize,
+    and prints them in console.
+
+    :param all_data: dict, traces data collected during run of algorithms. For more details, see
+                    'evaluate_optimizer' function.
+    :return: nothing
+    """
+
+    average_for_algo = {}
+
+    for Model in all_data:
+        for Dataset in all_data[Model]:
+            for Algorithm in all_data[Model][Dataset]:
+                data = all_data[Model][Dataset][Algorithm]
+
+                if not Algorithm in average_for_algo:
+                    average_for_algo[Algorithm] = []
+
+                objs = [[v[1] for v in d] for d in data] # leave only objective values
+                # here objs is a 2d list, where first dimension corresponds to
+                # particular repeat of experiment, and second dimension corresponds to index
+                # of optimization step during optimization
+                average_for_algo[Algorithm].append(objs)
+
+    # calculate averages
+    for Algorithm in average_for_algo:
+        mean_obj_vals = np.mean(average_for_algo[Algorithm], axis=0)
+        minimums = np.min(mean_obj_vals, axis=1)
+        f_calls = np.argmin(mean_obj_vals, axis=1)
+
+        min_mean = np.mean(minimums)
+        min_stdd = np.std(minimums)
+        min_best = np.min(minimums)
+
+        f_mean = np.mean(f_calls)
+        f_stdd = np.std(f_calls)
+        f_best = np.min(f_calls)
+
+        def fmt(float_value):
+            return ("%.3f" % float_value)
+
+        output = "|".join([fmt(min_mean) + " +/- " + fmt(min_stdd)] + [fmt(v) for v in [ min_best, f_mean, f_stdd, f_best]])
+        result = table_template + output
+        print("")
+        print(Algorithm)
+        print(result)
+
+
+def evaluate_optimizer((base_estimator, model, dataset, max_iter, seed_value)):
```

Comment:
I made changes so that code runs with python 3.4. Let me know if this still raises errors.